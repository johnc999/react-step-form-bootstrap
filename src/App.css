.container {
  height: 100%;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
}

.page_header {
  min-width: 650px;
  padding-top: 50px;
}

.progress_bar_padding {
  padding-top: 40px;
}

#page_2_continue {
  margin-left: -25px;
}
