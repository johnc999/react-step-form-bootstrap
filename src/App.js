import React from 'react';
import Form from './components/Form';
import './App.css';

function App() {
  return (
    <div className="App">
      <div className="container">
        <Form />
      </div>
    </div>
  );
}

export default App;
