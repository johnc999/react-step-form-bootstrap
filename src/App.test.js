import React from 'react';
import { cleanup, render, fireEvent, getByTestId, screen } from '@testing-library/react';
import App from './App';
import '@testing-library/jest-dom'

afterEach(cleanup);

test('First Name input exists', () => {
  const { getByText } = render(<App />);
  const inputElement = screen.getByTestId('firstname-input');
  const jsdomAlert = window.alert;  // remember the jsdom alert
  window.alert = () => {};  // provide an empty implementation for window.alert
  expect(inputElement).toBeInTheDocument();
});

test('App matches snapshot', () => {
  const { asFragment } = render(<App />)
  const jsdomAlert = window.alert;  // remember the jsdom alert
  window.alert = () => {};  // provide an empty implementation for window.alert
  expect(asFragment(<App />)).toMatchSnapshot()
});

test("page 1 bad email",async()=>{
  const nextStep = jest.fn();
  const { getByLabelText, getByText } = render(<App nextStep={nextStep} />);

  fireEvent.change(screen.getByTestId('firstname-input'), { target: { value: 'Jerry' } });
  fireEvent.change(screen.getByTestId('lastname-input'), { target: { value: 'Jones' } });
  fireEvent.change(screen.getByTestId('email-input'), { target: { value: 'jerry.jones.com' } });

  const btnIncrement = screen.getByTestId("page-1-continue-btn");
  fireEvent.click(btnIncrement);

  const jsdomAlert = window.alert;  // remember the jsdom alert
  window.alert = () => {};  // provide an empty implementation for window.alert

  const streetno = screen.queryByTestId('streetnumber-input');
  expect(streetno).not.toBeInTheDocument();
});

test("page 1 success",async()=>{
  const nextStep = jest.fn();
  const { getByLabelText, getByText } = render(<App nextStep={nextStep} />);

  fireEvent.change(screen.getByTestId('firstname-input'), { target: { value: 'Jerry' } });
  fireEvent.change(screen.getByTestId('lastname-input'), { target: { value: 'Jones' } });
  fireEvent.change(screen.getByTestId('email-input'), { target: { value: 'jerry.jones@aabb.com' } });

  const btnIncrement = screen.getByTestId("page-1-continue-btn");
  fireEvent.click(btnIncrement);

  const jsdomAlert = window.alert;  // remember the jsdom alert
  window.alert = () => {};  // provide an empty implementation for window.alert

  const streetno = screen.queryByTestId('streetnumber-input');
  expect(streetno).toBeInTheDocument();
});

test("page 2 bad street number",async()=>{
  const nextStep = jest.fn();
  const { getByLabelText, getByText } = render(<App nextStep={nextStep} />);

  fireEvent.change(screen.getByTestId('firstname-input'), { target: { value: 'Jerry' } });
  fireEvent.change(screen.getByTestId('lastname-input'), { target: { value: 'Jones' } });
  fireEvent.change(screen.getByTestId('email-input'), { target: { value: 'jerry.jones@aabb.com' } });

  const btnIncrement = screen.getByTestId("page-1-continue-btn");
  fireEvent.click(btnIncrement);

  const jsdomAlert = window.alert;  // remember the jsdom alert
  window.alert = () => {};  // provide an empty implementation for window.alert

  const streetno = screen.queryByTestId('streetnumber-input')
  expect(streetno).toBeInTheDocument()

  fireEvent.change(screen.getByTestId('streetnumber-input'), { target: { value: 'abc' } });
  fireEvent.change(screen.getByTestId('streetname-input'), { target: { value: 'Queen' } });
  fireEvent.change(screen.getByTestId('streettype-input'), { target: { value: 'St' } });
  fireEvent.change(screen.getByTestId('suburb-input'), { target: { value: 'Brisbane' } });
  fireEvent.change(screen.getByTestId('postcode-input'), { target: { value: '4000' } });

  const btnIncrement2 = screen.getByTestId("page-2-continue-btn");
  fireEvent.click(btnIncrement2);

  const page3backbutton = screen.queryByTestId('p3-back-btn');
  expect(page3backbutton).not.toBeInTheDocument();

});

test("page 2 success",async()=>{
  const nextStep = jest.fn();
  const { getByLabelText, getByText } = render(<App nextStep={nextStep} />);

  fireEvent.change(screen.getByTestId('firstname-input'), { target: { value: 'Jerry' } });
  fireEvent.change(screen.getByTestId('lastname-input'), { target: { value: 'Jones' } });
  fireEvent.change(screen.getByTestId('email-input'), { target: { value: 'jerry.jones@aabb.com' } });

  const btnIncrement = screen.getByTestId("page-1-continue-btn");
  fireEvent.click(btnIncrement);

  const jsdomAlert = window.alert;  // remember the jsdom alert
  window.alert = () => {};  // provide an empty implementation for window.alert

  const streetno = screen.queryByTestId('streetnumber-input')
  expect(streetno).toBeInTheDocument()

  fireEvent.change(screen.getByTestId('streetnumber-input'), { target: { value: '33' } });
  fireEvent.change(screen.getByTestId('streetname-input'), { target: { value: 'Queen' } });
  fireEvent.change(screen.getByTestId('streettype-input'), { target: { value: 'St' } });
  fireEvent.change(screen.getByTestId('suburb-input'), { target: { value: 'Brisbane' } });
  fireEvent.change(screen.getByTestId('postcode-input'), { target: { value: '4000' } });

  const btnIncrement2 = screen.getByTestId("page-2-continue-btn");
  fireEvent.click(btnIncrement2);

  const page3backbutton = screen.queryByTestId('p3-back-btn');
  expect(page3backbutton).toBeInTheDocument();

});

