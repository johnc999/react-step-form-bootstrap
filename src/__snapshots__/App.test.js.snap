// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="container"
    >
      <div
        class="form-container page_header"
      >
        <h2
          class="mb-5"
        >
          Personal Details
        </h2>
        <div
          class="progress"
        >
          <div
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="0"
            class="progress-bar"
            role="progressbar"
            style="width: 0%;"
          >
            0/2 Steps Completed
          </div>
        </div>
        <div
          class="form-group progress_bar_padding"
        >
          <label
            for="firstname"
          >
            First Name:
          </label>
          <input
            class="form-control"
            data-testid="firstname-input"
            name="firstname"
            type="text"
            value=""
          />
        </div>
        <div
          class="form-group"
        >
          <label
            for="lastname"
          >
            Last Name:
          </label>
          <input
            class="form-control"
            data-testid="lastname-input"
            name="lastname"
            type="text"
            value=""
          />
        </div>
        <div
          class="form-group"
        >
          <label
            for="email"
          >
            Email:
          </label>
          <input
            class="form-control"
            data-testid="email-input"
            name="email"
            type="text"
            value=""
          />
        </div>
        <div
          class="form-group"
        >
          <label
            for="phone"
          >
            Phone:
          </label>
          <input
            class="form-control"
            data-testid="phone-input"
            name="phone"
            type="text"
            value=""
          />
        </div>
        <br />
        <div
          class="text-right"
        >
          <button
            class="btn btn-primary"
            data-testid="page-1-continue-btn"
          >
            Continue
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
